packages/m2m-gateway/test/integration/delegations/getProducerDelegations.test.ts

```diff
@@ -0,0 +1,146 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { delegationApi, m2mGatewayApi } from "pagopa-interop-api-clients";
+import { generateId } from "pagopa-interop-models";
+import { getMockedApiDelegation } from "pagopa-interop-commons-test";
+import {
+  delegationService,
+  expectApiClientGetToHaveBeenCalledWith,
+  mockInteropBeClients,
+} from "../../integrationUtils.js";
+import { PagoPAInteropBeClients } from "../../../src/clients/clientsProvider.js";
+import { getMockM2MAdminAppContext } from "../../mockUtils.js";
+import { WithMaybeMetadata } from "../../../src/clients/zodiosWithMetadataPatch.js";
+import { unexpectedDelegationKind } from "../../../src/model/errors.js";
+
+describe("getProducerDelegations", () => {
+  const mockParams: m2mGatewayApi.GetProducerDelegationsQueryParams = {
+    states: ["WAITING_FOR_APPROVAL"],
+    eserviceIds: [generateId()],
+    delegateIds: [generateId()],
+    delegatorIds: [generateId()],
+    offset: 0,
+    limit: 10,
+  };
+
+  const mockApiDelegation1 = getMockedApiDelegation({
+    kind: delegationApi.DelegationKind.Values.DELEGATED_PRODUCER,
+  });
+
+  const mockApiDelegation2 = getMockedApiDelegation({
+    kind: delegationApi.DelegationKind.Values.DELEGATED_PRODUCER,
+  });
```

Comment:
Since you're here, can you apply the same fix to `getConsumerDelegations.test.ts` to align it with this? 
I'm talking about this fix to remove `getMockWithMetadata` https://github.com/pagopa/interop-be-monorepo/pull/2038#discussion_r2163490402